app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/BatteryNormalize.java

```diff
@@ -0,0 +1,44 @@
+package nodomain.freeyourgadget.gadgetbridge.util;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+import nodomain.freeyourgadget.gadgetbridge.deviceevents.GBDeviceEventBatteryInfo;
+
+/**
+ * Created by Raziel on 13/04/2017.
+ */
+
+public class BatteryNormalize {
+    private int counter = 0;
+    private int levelMid = 0;
+    private GregorianCalendar lastCheck = new GregorianCalendar();
+    private static int INTERVAL_SECOND_CHECK  = 30;
+    private static int MAX_COUNT_CHECK = 6;
+
+    public void BatteryNormalize(int interval,int numerberCheck){
+        INTERVAL_SECOND_CHECK=interval;
+        MAX_COUNT_CHECK=numerberCheck;
+    }
+
+    //use this function if the battery level of the device oscillates
+    public short getLevelNormalize(int data, GBDeviceEventBatteryInfo cmd) {
```

Comment:
I try to explain it without log ... it often happens that the battery level drops for a few seconds or minutes and then goes up, the difference ranges between 2 and 20 percent so I thought I would spot 6 values ​​with a 30 second interval where recovery Only the most other value at the end of the 6 controls I return the highest value that is the correct one, so I understand if the battery has actually been discharged or charged or you have temporarily lowered the battery level